app/federation-v2/server/matrix-client/user.ts

```diff
@@ -16,6 +17,21 @@ interface ICreateUserResult {
 export const invite = async (inviterId: string, roomId: string, invitedId: string): Promise<void> => {
 	console.log(`[${inviterId}-${invitedId}-${roomId}] Inviting user ${invitedId} to ${roomId}...`);
 
+	// Find the inviter user
+	let inviterUser = MatrixBridgedUser.getById(inviterId);
+	// Get the user
+	const user = await dataInterface.user(inviterId);
+
+	// The inviters user doesn't yet exist in matrix
+	if (!inviterUser) {
+		console.log(`[${inviterId}-${invitedId}-${roomId}] Creating remote inviter user...`);
+
+		// Create the missing user
+		inviterUser = await matrixClient.user.createRemote(user);
+
+		console.log(`[${inviterId}-${invitedId}-${roomId}] Inviter user created as ${inviterUser.mui}...`);
+	}
```

Comment:
I moved this up.  This makes sure the user thats inviting is created first in the matrix side.  This removed a lot of errors down below